### ISSUE TITLE
Cuda synchronize issue

### ISSUE THREAD

**Issue by crockpotveggies** [eclipse/deeplearning4j#6892]
After running training in the MNISTAnomalyExample the evaluation code triggers a `cudaStreamSynchronize` failure.

Running `1.0.0_changes` branch of examples. Ubuntu 18, snapshots, CUDA 10. Compiled locally.

https://github.com/deeplearning4j/dl4j-examples/blob/8f4224c9c803394e287accb0f84fe9b624d923d5/dl4j-examples/src/main/java/org/deeplearning4j/examples/feedforward/anomalydetection/MNISTAnomalyExample.java#L105

```
o.d.o.l.ScoreIterationListener - Score at iteration 14998 is 0.10566424120426078
o.d.o.l.ScoreIterationListener - Score at iteration 14999 is 0.10088278996497956
Epoch 29 complete
CUDA error at /mnt/jenkins/workspace/deeplearning4j-master-linux-x86_64-cuda-10.0/libnd4j/blas/cuda/NativeOps.cu:1861 code=77(<unknown>) "dZ"
CUDA error at /mnt/jenkins/workspace/deeplearning4j-master-linux-x86_64-cuda-10.0/libnd4j/blas/cuda/NativeOps.cu:1861 code=77(<unknown>) "dZ"
Exception in thread "main" java.lang.RuntimeException: cudaStreamSynchronize(...) failed
	at org.nd4j.nativeblas.Nd4jCuda$NativeOps.streamSynchronize(Native Method)
	at org.nd4j.jita.allocator.pointers.cuda.cudaStream_t.synchronize(cudaStream_t.java:38)
	at org.nd4j.jita.flow.impl.SynchronousFlowController.commitTransfer(SynchronousFlowController.java:314)
	at org.nd4j.jita.constant.ProtectedCudaConstantHandler.moveToConstantSpace(ProtectedCudaConstantHandler.java:228)
	at org.nd4j.jita.constant.CudaConstantHandler.moveToConstantSpace(CudaConstantHandler.java:42)
	at org.nd4j.jita.allocator.impl.AtomicAllocator.moveToConstant(AtomicAllocator.java:1070)
	at org.nd4j.jita.allocator.tad.DeviceTADManager.getTADOnlyShapeInfo(DeviceTADManager.java:113)
	at org.nd4j.linalg.jcublas.ops.executioner.CudaExecutioner.naiveExec(CudaExecutioner.java:229)
	at org.nd4j.linalg.jcublas.ops.executioner.CudaExecutioner.exec(CudaExecutioner.java:526)
	at org.nd4j.linalg.api.ndarray.BaseNDArray.sum(BaseNDArray.java:4711)
	at org.nd4j.linalg.api.ndarray.BaseNDArray.sumNumber(BaseNDArray.java:1344)
	at org.nd4j.linalg.lossfunctions.impl.LossL2.computeScore(LossL2.java:108)
	at org.nd4j.linalg.lossfunctions.impl.LossMSE.computeScore(LossMSE.java:59)
	at org.deeplearning4j.nn.layers.BaseOutputLayer.computeScore(BaseOutputLayer.java:90)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.scoreHelper(MultiLayerNetwork.java:2477)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.score(MultiLayerNetwork.java:2432)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.score(MultiLayerNetwork.java:2421)
	at org.deeplearning4j.examples.feedforward.anomalydetection.MNISTAnomalyExample.main(MNISTAnomalyExample.java:105)
	Suppressed: java.lang.RuntimeException: cudaStreamSynchronize(...) failed
		at org.nd4j.nativeblas.Nd4jCuda$NativeOps.streamSynchronize(Native Method)
		at org.nd4j.linalg.jcublas.context.CudaContext.syncOldStream(CudaContext.java:118)
		at org.nd4j.linalg.jcublas.context.CudaContext.syncOldStream(CudaContext.java:108)
		at org.nd4j.linalg.jcublas.ops.executioner.CudaExecutioner.commit(CudaExecutioner.java:1884)
		at org.nd4j.linalg.memory.abstracts.Nd4jWorkspace.close(Nd4jWorkspace.java:601)
		at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.scoreHelper(MultiLayerNetwork.java:2478)
		... 3 more

Process finished with exit code 1
```

Aha! Link: https://skymindai.aha.io/features/ND4J-140

**Comment by raver119** [comment]
Is cuDNN used?

**Comment by crockpotveggies** [comment]
@raver119 yes

**Comment by AlexDBlack** [comment]
CuDNN isn't used in that example. It's dense layers and output layer only, nothing special.

**Comment by raver119** [comment]
In my case cuDNN was used, basic LSTM nn, but crash happened only after multiple evaluations. So it's safe to rule out cuDNN i think then.

**Comment by crockpotveggies** [comment]
Correct I've added CuDNN locally to speed up example runtime for dev purposes.

**Comment by ecthx** [comment]
I'm also being affected by this issue. 
Running snapshots rl4j A3C with CUDA10 compiled locally.
Thanks

`o.d.r.l.a.AsyncThread - Thread crashed: null
CUDA error at /mnt/jenkins/workspace/deeplearning4j-master-linux-x86_64-cuda-10.0/libnd4j/blas/cuda/NativeOps.cu:1895 code=77(<unknown>) "dZ" 
Exception in thread "UniGC thread 2" Exception in thread "UniGC thread 5" Exception in thread "UniGC thread 3" Exception in thread "UniGC thread 4" Exception in thread "UniGC thread 0" Exception in thread "UniGC thread 1" java.lang.RuntimeException: cudaStreamSynchronize(...) failedo.d.r.l.a.AsyncThread - Thread crashed: null

	at org.nd4j.nativeblas.Nd4jCuda$NativeOps.streamSynchronize(Native Method)
	at org.nd4j.linalg.jcublas.context.CudaContext.syncOldStream(CudaContext.java:118)
	at org.nd4j.linalg.jcublas.context.CudaContext.syncOldStream(CudaContext.java:108)
	at org.nd4j.linalg.jcublas.ops.executioner.CudaExecutioner.commit(CudaExecutioner.java:1899)
	at org.nd4j.linalg.memory.abstracts.Nd4jWorkspace.close(Nd4jWorkspace.java:602)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.outputOfLayerDetached(MultiLayerNetwork.java:1286)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.output(MultiLayerNetwork.java:2318)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.output(MultiLayerNetwork.java:2281)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.output(MultiLayerNetwork.java:2272)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.output(MultiLayerNetwork.java:2259)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.output(MultiLayerNetwork.java:2352)
	at org.deeplearning4j.rl4j.network.ac.ActorCriticSeparate.outputAll(ActorCriticSeparate.java:78)
	at org.deeplearning4j.rl4j.policy.ACPolicy.nextAction(ACPolicy.java:77)
	at org.deeplearning4j.rl4j.policy.ACPolicy.nextAction(ACPolicy.java:39)
	at org.deeplearning4j.rl4j.learning.async.AsyncThreadDiscrete.trainSubEpoch(AsyncThreadDiscrete.java:94)
	at org.deeplearning4j.rl4j.learning.async.AsyncThread.run(AsyncThread.java:103)
java.lang.RuntimeException: cudaStreamSynchronize(...) failed
`

**Comment by ecthx** [comment]
Rl4J A3C runs ok now with CUDA backend.
Sorry for not opening a separate issue for this.
